app/indexers/identity_metadata_datastream_indexer.rb

```diff
@@ -3,45 +3,59 @@
 class IdentityMetadataDatastreamIndexer
   include SolrDocHelper
 
-  attr_reader :resource
+  attr_reader :cocina_object
 
-  def initialize(resource:, **)
-    @resource = resource
+  def initialize(cocina:, **)
+    @cocina_object = cocina
   end
 
   # @return [Hash] the partial solr document for identityMetadata
   def to_solr
-    solr_doc = {}
-    solr_doc['objectType_ssim'] = resource.identityMetadata.objectType
-
-    plain_identifiers = []
-    ns_identifiers = []
-    if source_id.present?
-      (name, id) = source_id.split(/:/, 2)
-      plain_identifiers << id
-      ns_identifiers << source_id
-      solr_doc['source_id_ssim'] = [source_id]
-    end
+    {
+      'objectType_ssim' => [object_type],
+      'dor_id_tesim' => [source_id_value, barcode, catkey].compact,
+      'identifier_ssim' => prefixed_identifiers,
+      'identifier_tesim' => prefixed_identifiers,
```

Comment:
Not your doing, but just to point out it is odd to put identifiers in a `*_tesim` (english tokenized text) field.